sdk/core/core-http/lib/credentials/tokenCredential.ts

```diff
@@ -52,5 +49,8 @@ export interface AccessToken {
  * @param credential The assumed TokenCredential to be tested.
  */
 export function isTokenCredential(credential: any): credential is TokenCredential {
+  if (!credential) {
+    return false;
+  }
```

Comment:
@daviwil Here we check if `getToken` exists in `credential`. This would return `true` if the given object had a property and not function by the name `getToken`. How about using 
```javascript
return credential && typeof credential.getToken === "function"
```

Reply:
Good point, the change looks good to me!

Reply:
Updated!!